test/login.test.ts

```diff
@@ -0,0 +1,221 @@
+import { expect } from 'chai';
+import axios from 'axios';
+import server from '../src/index';
+import { prisma } from '../src/app/client/client';
+import 'dotenv/config';
+import bcrypt from 'bcrypt';
+
+describe('login mutation', () => {
+  const port = process.env.PORT;
+  const url = `http://localhost:${port}/`;
+  before(async () => {
+    try {
+      await server.listen(port).then(async ({ url }) => {
+        console.log(url);
+      });
+    } catch (error) {
+      console.log(error.message);
+    }
+
+    try {
+      await prisma.$connect();
+    } catch (error) {
+      console.log(error);
+    }
+    console.log(`listening on ${url}`);
+    console.log('connected to testdb');
+
+    await prisma.user.create({
+      data: {
+        name: 'Sam',
+        email: 'sam@example.com',
+        password: await bcrypt.hash('Sam123', 10),
+        birthDate: '09-04-2004',
+      },
+    });
+  });
+
+  after(async () => {
+    if (server) {
+      await server.stop();
+      console.log('Server stopped');
+    }
+
+    await prisma.user.deleteMany();
+
+    if (prisma) {
+      await prisma.$disconnect();
+      console.log('Database testdb disconnected');
+    }
+  });
+
+  it('should return an error for invalid email', async () => {
+    const mutation = `
+            mutation login ($data: LoginInput!) {
+                login (data: $data) {
+                    user {
+                        id,
+                        name,
+                        email,
+                        birthDate
+                    },
+                    token
+                }
+            }
+        `;
```

Comment:
essa variável se repete em todos os casos de teste, não daria pra extrair ela de cada `it`?